clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java

```diff
@@ -159,24 +159,23 @@ private static ChannelBuilder create(SecurityProtocol securityProtocol,
     }
 
     // Visibility for testing
+    @SuppressWarnings("unchecked")
     protected static Map<String, Object> channelBuilderConfigs(final AbstractConfig config, final ListenerName listenerName) {
-        Map<String, ?> parsedConfigs;
+        Map<String, Object> parsedConfigs;
```

Comment:
I'm confused, why does convert the type of config will change the way we interpret the config value?

Reply:
> Does it make sense to just change the logging

Pardon me, I failed to get your point.

> And in terms of misleading, does this log confuse the user by any chance?

for example, the ssl-related configs used to create ssl protocol are viewed as *unknown* when creating ```KafkaAdmin```.

> why does convert the type of config will change the way we interpret the config value?

the fix is unrelated to the type of config. The ```configs``` returned by this method is changed to the inner map (```RecordingMap```) of ```AbstractConfig``` so the keys used to call ```Map#get``` will be added to *used list* of ```AbstractConfig```.